tsl/src/fdw/relinfo.c

```diff
@@ -85,35 +85,18 @@ apply_fdw_and_server_options(TsFdwRelInfo *fpinfo)
 TsFdwRelInfo *
 fdw_relinfo_get(RelOptInfo *rel)
```

Comment:
nit: since this function is now used instead of alloc, maybe it makes sense to pass `TsFdwRelInfoType reltyp` as well? If private data is not exists set it, if it exists maybe validate against the type? not sure how much it would be useful anyway, since you have to modify other fields as well

Reply:
I returned the alloc function back to enable more strict checks. Let's see if I broke something.